### DISCUSSION TITLE
Version 0.17.0 Release Candidate

### DISCUSSION THREAD

**Discussion by PrestonN** [FreeTubeApp/FreeTube#2309] [Announcements]
Hey all. Since there is a rather large amount of changes this time around, we'll be doing a release candidate for the upcoming v0.17.0 release. During this time, we'll be on the lookout for show stopping bugs and issues related to these new changes that might prevent us from doing a full release.

If anyone happens to have any spare time, please take a moment to try out this new release and report back here if you happen to encounter any issues with the new release. If no major issues are found, then we will do a full release shortly after this testing period.

When downloading a release candidate, this version of FreeTube will overwrite your existing version. While it's highly unlikely that issues will occur during this installation, please consider backing up your data when installing to prevent any data loss. If issues do happen, you'll be able to revert to v0.16.0 by installing the builds from our website, and installing them over top of the release candidate.

Download the release candidate [here](https://github.com/FreeTubeApp/FreeTube/actions/runs/2484662180) (A GitHub account is required to download)

Changes / Patch Notes will be listed below.

**Comment by PrestonN** [comment]
Current changlog since v0.16.0. Styling / formatting are subject to change between now and the full release.

### Changes
* Fix iconsistent white icon by @vallode in https://github.com/FreeTubeApp/FreeTube/pull/2088
* Fix thumbnail being obstructed by time tooltip by @0x6e656b6f in https://github.com/FreeTubeApp/FreeTube/pull/2078
* Sync theme between windows by @vallode in https://github.com/FreeTubeApp/FreeTube/pull/2090
* Fix/SearchHistoryDuplicate by @hockerschwan in https://github.com/FreeTubeApp/FreeTube/pull/2124
* Fix randomly occurring error when searching by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2148
* Fast forward = Playback rate × Interval by @hockerschwan in https://github.com/FreeTubeApp/FreeTube/pull/2153
* Only fetch the most recent update from the GitHub releases by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2169
* Switch debug and picture in picture shortcuts for consistency by @vallode in https://github.com/FreeTubeApp/FreeTube/pull/2141
* Fix search input shortcut by @vallode in https://github.com/FreeTubeApp/FreeTube/pull/2140
* Fix: trending cache by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/2125
* Hide filter on history and playlist page when page is empty by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2177
* Add the alternate search keyboard shortcuts COMMAND+L on MacOS and CTRL+L on other platforms by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2175
* Pr issue dependent by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/1832
* Auto label the PR when submitted by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2183
* Auto Remove outdated labels by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2186
* Auto Label PR changes requested by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2184
* Auto PR Label Conflicts by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2185
* test workflow API rate limit exceeded by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2188
* Auto label duplicate issues by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2187
* Delete dependent-issues.yml by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2198
* testing workflow by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2201
* revert workflows by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2202
* Allow DASH formats for livestream replays by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2208
* Clear watch progress from view cache when removing video form history by @Svallinn in https://github.com/FreeTubeApp/FreeTube/pull/2220
* Hide Picture in Picture button and disable keyboard shortcut for audio formats by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2227
* Update all GH workflow config to update node 14.x > 16.x by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/1898
* System theme preference by @vallode in https://github.com/FreeTubeApp/FreeTube/pull/1800
* channel fix (banner, playlist) by @hockerschwan in https://github.com/FreeTubeApp/FreeTube/pull/2215
* Prevent search input reset from overwriting clipboard on Linux by @vallode in https://github.com/FreeTubeApp/FreeTube/pull/2108
* Fix #1774 - Dropdown doesn't match theme in dark mode by @jilv220 in https://github.com/FreeTubeApp/FreeTube/pull/2199
* Update build script to build ARM64 version on Apple Silicon machines by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/1918
* Update invidious-instances.json by @efb4f5ff-1298-471a-8973-3d47447115dc in https://github.com/FreeTubeApp/FreeTube/pull/2236
* Feat: Channel Member Icons + Load Custom Channel Emojis in Comments by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/1819
* Fix the stats for nerds modal hogging all the keyboard events by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2253
* Add support for next/previous/pause/resume global hotkeys/actions by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2239
* Fix: csv import/export by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/2247
* Fix: wrong info displayed when quickly navigate from a channel to another by @hockerschwan in https://github.com/FreeTubeApp/FreeTube/pull/2259
* Fix typo by @maxnordlund in https://github.com/FreeTubeApp/FreeTube/pull/2267
* Fix loading subtitles for non-English languages by @hockerschwan https://github.com/FreeTubeApp/FreeTube/pull/2268
* More SponsorBlock features implemented by @ChunkyProgrammer https://github.com/FreeTubeApp/FreeTube/pull/1849
* README: Warning for browser extension by @efb4f5ff-1298-471a-8973-3d47447115dc https://github.com/FreeTubeApp/FreeTube/pull/2210
* Issue Templates updated by @efb4f5ff-1298-471a-8973-3d47447115dc https://github.com/FreeTubeApp/FreeTube/pull/2257
* README: winget package manager added by @efb4f5ff-1298-471a-8973-3d47447115dc https://github.com/FreeTubeApp/FreeTube/pull/2143
* Dividers added to "more dropdown menu" by @PikachuEXE https://github.com/FreeTubeApp/FreeTube/pull/1748
* Feature: Screenshot of video by @hockerschwan in https://github.com/FreeTubeApp/FreeTube/pull/2221
* Fix handling of youtube link middle mouse click by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2083
* Even spacing on top-nav by @vallode https://github.com/FreeTubeApp/FreeTube/pull/2280
* Clean up ft-input for top-nav by @vallode https://github.com/FreeTubeApp/FreeTube/pull/2279 
* Fix webpack deprecation warnings in dev-runner by @absidue https://github.com/FreeTubeApp/FreeTube/pull/2282
### Dependency Updates
* Bump follow-redirects from 1.14.7 to 1.14.8 by @dependabot in https://github.com/FreeTubeApp/FreeTube/pull/2079
* ^ Update electron 16.x by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2082
* Bump minimist from 1.2.5 to 1.2.6 by @dependabot in https://github.com/FreeTubeApp/FreeTube/pull/2151
* Bump plist from 3.0.4 to 3.0.5 by @dependabot in https://github.com/FreeTubeApp/FreeTube/pull/2150
* Bump node-forge from 1.0.0 to 1.3.0 by @dependabot in https://github.com/FreeTubeApp/FreeTube/pull/2147
* Update electron to 16.2.0 by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2117
* update yt-channel-info by @ChunkyProgrammer in https://github.com/FreeTubeApp/FreeTube/pull/2126
* Update electron to 16.2.2 by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2168
* Bump marked and electron-context-menu by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2176
* Bump autolinker and forge to latest versions by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2200
* Update electron to 16.2.5 by @PikachuEXE in https://github.com/FreeTubeApp/FreeTube/pull/2212
* Bump ejs from 3.1.6 to 3.1.7 by @dependabot in https://github.com/FreeTubeApp/FreeTube/pull/2233
* Update marked to 4.0.14 and ytsr to 3.8.0 by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2229
* Bump Babel to latest version and update Babel targets by @absidue in https://github.com/FreeTubeApp/FreeTube/pull/2182

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Things to address 

- [x] fixed in https://github.com/FreeTubeApp/FreeTube/pull/2310 - invalid characters dont get replaced automatically resulting in failed download (https://youtu.be/kVwkLb8zxq0). If u have enabled 'ask for download path' Windows will say to u it is not possible to use such a file format but if u choose to download videos automatically to a directory then it will try to do that.
- [x] fixed in https://github.com/FreeTubeApp/FreeTube/pull/2310 - Test the following on [this video](https://youtu.be/lEaPQ7YrWQg). When u download this and u have the setting enabled 'ask for download path' it will finish the download successfully. Change the setting so u have a default download folder and u see that download will fail (if it matters at all path is `C:\Users\x\Desktop`). Maybe related to https://github.com/FreeTubeApp/FreeTube/issues/2110
- [x] fixed in https://github.com/FreeTubeApp/FreeTube/pull/2310 - when 'ask for download path' is enabled file name got cut of in this video https://youtu.be/mSf5IeiQVpY
![Capture](https://user-images.githubusercontent.com/73130443/173255287-310f0ffb-6d1b-42e0-b233-cca6d29a36b9.PNG)

- [x] fixed in https://github.com/FreeTubeApp/FreeTube/pull/2425  - cant scroll down on download menu. It will first scroll down the recommended videos section and then it will scroll down on the download menu. This is also the case when u have enabled comments. it will scroll through the recommended videos first and then it will go to the download section. i'm using a touchpad

https://user-images.githubusercontent.com/73130443/173255376-6e9560b9-4c4b-4d5f-b1c4-dd1793a85352.mp4

- [x] fixed in #2356 - https://github.com/FreeTubeApp/FreeTube/issues/2343
- [x]  fixed in #2354 - search something -> see search suggestions -> navigate through the suggestions with the arrow keys on your key board -> get stuck

https://user-images.githubusercontent.com/73130443/174895140-99d3af22-8e73-48ea-926b-efc17aabad5d.mp4

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
If u want to download the latest version of the RC branch go to https://github.com/FreeTubeApp/FreeTube/actions/workflows/build.yml?query=branch%3Av0.17.0-RC

**Comment by Holy-Person** [comment]
When is this planned to release as a full version? Any rough time planned or no real date yet?

**Reply by Holy-Person** [reply]
I meant as full 0.17 not 1.0

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
We are still in the progress of ironing bugs out. So there is no date planned for v0.17.0

**Comment by ghost** [comment]
I just downloaded 0.17 beta and found that playlists are not yet implemented. Have playlists been postponed to 0.18?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Playlists are indeed postponed

**Comment by DI555** [comment]
@PrestonN , firstly, let me say great thanks for the FreeTube app!

but still we have a lots of machines with Win7 32 bits((.
could you please provide a Win32 builds (at least portable one) versions?
is it possible?

**Reply by PrestonN** [reply]
Hey there. This isn't possible anymore unfortunately.

Electron dropped support for Windows 7 as well as 32 bit quite a while ago. Windows 7 has also reached end of life from Microsoft. If Microsoft isn't supporting the OS anymore, then there isn't much of a reason for us to attempt to support it either.

I would highly recommend using a different operating system if possible.

**Comment by PrestonN** [comment]
Locking discussion as v0.17.0 has been released. Feel free to start a new discussion over [here](https://github.com/FreeTubeApp/FreeTube/discussions/2435).
